00xx-vinegar.md

```diff
@@ -0,0 +1,101 @@
+    - Now, looking into `verifier.rs/to_batch` -- after `OraclesResult` that corresponds to the previous computation, we compute `f_comm`, and start collecting evaluations, now into `evaluations`, to return them together with previous data in `BatchEvaluationProof`.
+    - Note: `prev_challenges` that are the input to `verifier.rs` are coming from 2 itertaions before --- not from the previous proof, but from the one before that. Hence they are over the same group (`prev.challenges.comm` is a commitment, that is group element, and `prev_challenges.chals` are scalar field elements) as the target one for the proof that we are verifying. Hence `prev_challenges.chals` are absorbed in this step.
+    - volhovm: what is this "checking that the evaluations satisfy the combined constraint polynomial"? I don't think there's any *checking* or *verification* happening at this point, we merely compute thes polynomial, no?
+1. Verify that the polynomial commitments and their evaluations  are consistent by checking the 'opening proof' of the polynomial commitment scheme.
+    - This corresponds to `OpeningProof::verify` call in the end of `batch_verify`. Internally, this will combine all the evaluations together, and check that they are consistent with the commitments, jointly.
+
+In order to *recursively* verify a kimchi proof in pickles, we need to be able to efficiently run these steps inside a circuit. However, some of the operations are run on elliptic curve points -- represented as tuples in the *base field* -- and others are run against the *scalar field*, that is the field $F_q$ of scalars for elliptic curve points that form a group. To handle both without an expensive simulation of foreign field arithmetic in the verifier circuit, instead we generate a pair of circuits, where each shares some of the work.
+
+Pickles is our current implementation of a recursive verifier for the kimchi proof system, implemented as a pair of circuits Step and Wrap implemented on top of kimchi.
+
+The Step circuit is a kimchi proof over the Vesta curve (its commitments are Vesta curve points); the Wrap circuit is a kimchi proof over the Pallas curve.
+
+### Verifying a proof in a circuit
+
+Assume that we are verifying a Pallas (wrap) proof, but the below applies equivalently to Vesta. Here is how the list in the previous section maps on our curves:
+
+1. The logic for absorbing the commitments is run inside a Vesta proof.
+    - Wrap commitments are Pallas points made of two Pallas base field elements. These are Vesta scalar field elements, which makes them easy to reason about in the Vesta proof -- the next Step proof that verifies a Wrap.
```

Comment:
This is a weird place to intro that `vesta_{scalar,base} = pallas_{base,scalar}`. Might be worth extracting to a dedicated paragraph, whose gist is:
* pallas and vesta are a cycle of curves: the scalar field of vesta is the base field of pallas, and the scalar field of pallas is the base field of vesta
* pickles Wrap proofs use the pallas curve, and witness values use the pallas scalar field / vesta base field
* pickles Step proofs use the vesta curve, and witness values use the vesta scalar field / pallas base field.

Then, this can be simplified to
```suggestion
    - The commitments are formed of Vesta scalar field elements, so we can use native arithmetic.
```
or some such.

Reply:
Will create a separate paragraph. This was not intended to introduce the concept of curves. I'm just repeating the same thing several times in different phrasings/levels of elaboration which is supposed to help the reader to double-check there's no mistake (because everything is symmetric).